### PR TITLE
[Feature-3941][ui] Remove tenant name 

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
https://github.com/apache/incubator-dolphinscheduler/issues/3941

FEATURE-UI#3941: Removed Tenant Name

### CODE REVIEW

**Comment by CalvinKirs** [comment]
Hi, have you encountered any problems? You can leave a message here and I will help you solve it

**Comment by IamMujuziMoses** [comment]
@CalvinKirs yes, how do I run the Unit Test and e2e Tests Successfully?!

**Comment by IamMujuziMoses** [comment]
@CalvinKirs can I continue working on this or is it closed?!

**Comment by CalvinKirs** [comment]
> @CalvinKirs yes, how do I run the Unit Test and e2e Tests Successfully?!

The error message I saw that UT failed is:TenantMapperTest.testQueryTenantPaging:147 Values should be different. Actual: 0

Please check if your code is correct.

**Review by zhuangchong — COMMENTED** [review]
1.Please delete the attribute TENANT_NAME
dolphinscheduler-api/src/main/resources/i18n/messages.properties
dolphinscheduler-api/src/main/resources/i18n/messages_en_US.properties
dolphinscheduler-api/src/main/resources/i18n/messages_zh_CN.properties

2.Please delete the attribute tenantName
dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/TenantControllerTest.java
e2e/src/test/java/org/apache/dolphinscheduler/locator/security/TenantManageLocator.java 
e2e/src/test/java/org/apache/dolphinscheduler/page/security/TenantManagePage.java
e2e/src/test/resources/testData/security_zh_cn.yml

3.Please delete tenant_name in sql script
dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionMapper.xml

4.Please modify the tenant_name in the sql script to tenant_code
dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TenantMapper.xml
dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/UserMapper.xml 

5.Please modify the tenantName in the front end script
dolphinscheduler-ui/src/js/conf/home/pages/dag/_source/udp/_source/selectTenant.vue
dolphinscheduler-ui/src/js/conf/home/pages/security/pages/tenement/_source/createTenement.vue
dolphinscheduler-ui/src/js/conf/home/pages/security/pages/users/_source/createUser.vue 
dolphinscheduler-ui/src/js/conf/home/store/security/actions.js
dolphinscheduler-ui/src/js/module/i18n/locale/zh_CN.js

**Comment by xingchun-chen** [comment]
image
please change td[8] to td[7]

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java [hunk 1]
```diff
@@ -90,12 +88,11 @@
...
     public Result createTenant(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                @RequestParam(value = "tenantCode") String tenantCode,
-                               @RequestParam(value = "tenantName") String tenantName,
                                @RequestParam(value = "queueId") int queueId,
                                @RequestParam(value = "description", required = false) String description) throws Exception {
-        logger.info("login user {}, create tenant, tenantCode: {}, tenantName: {}, queueId: {}, desc: {}",
-                loginUser.getUserName(), tenantCode, tenantName, queueId, description);
-        Map<String, Object> result = tenantService.createTenant(loginUser, tenantCode, tenantName, queueId, description);
+        logger.info("login user {}, create tenant, tenantCode: {}, queueId: {}, desc: {}",
+                loginUser.getUserName(), tenantCode, queueId, description);
```
Comment (zhuangchong) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java:94]: logger.info("login user {}, create tenant, tenantCode: {}, queueId: {}, desc: {}",
                StringUtils.replaceNRTtoUnderline(loginUser.getUserName()),
                StringUtils.replaceNRTtoUnderline(tenantCode),
                queueId,
                StringUtils.replaceNRTtoUnderline(description));

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java [hunk 2]
```diff
@@ -177,12 +172,11 @@ public Result queryTenantlist(@ApiIgnore @RequestAttribute(value = Constants.SES
...
                                @RequestParam(value = "id") int id,
                                @RequestParam(value = "tenantCode") String tenantCode,
-                               @RequestParam(value = "tenantName") String tenantName,
                                @RequestParam(value = "queueId") int queueId,
                                @RequestParam(value = "description", required = false) String description) throws Exception {
-        logger.info("login user {}, updateProcessInstance tenant, tenantCode: {}, tenantName: {}, queueId: {}, description: {}",
-                loginUser.getUserName(), tenantCode, tenantName, queueId, description);
-        Map<String, Object> result = tenantService.updateTenant(loginUser, id, tenantCode, tenantName, queueId, description);
+        logger.info("login user {}, updateProcessInstance tenant, tenantCode: {}, queueId: {}, description: {}",
+                loginUser.getUserName(), tenantCode, queueId, description);
```
Comment (zhuangchong) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java:178]: logger.info("login user {}, updateProcessInstance tenant, tenantCode: {}, queueId: {}, description: {}",
                StringUtils.replaceNRTtoUnderline(loginUser.getUserName()),
                StringUtils.replaceNRTtoUnderline(tenantCode),
                queueId,
                StringUtils.replaceNRTtoUnderline(description));

**Review by zhuangchong — COMMENTED** [review]
Solve the sonar problem: Change this code to not log user-controlled data.

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/TenantServiceTest.java [hunk 3]
```diff
@@ -83,17 +83,17 @@ public void testCreateTenant() {
         try {
```
[the remark this answers is not in the document]
Comment (zhuangchong) [reply dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/service/TenantServiceTest.java:83]: private static final String tenantName = "TenantServiceTest"; 
The tenantName variable is not called and can be removed.

File: sql/dolphinscheduler_mysql.sql [hunk 4]
```diff
@@ -756,7 +756,6 @@ DROP TABLE IF EXISTS `t_ds_tenant`;
 CREATE TABLE `t_ds_tenant` (
   `id` int(11) NOT NULL AUTO_INCREMENT COMMENT 'key',
   `tenant_code` varchar(64) DEFAULT NULL COMMENT 'tenant code',
-  `tenant_name` varchar(64) DEFAULT NULL COMMENT 'tenant name',
```
Comment (zhuangchong) [inline sql/dolphinscheduler_mysql.sql:759]: dolphinscheduler-postgre.sql file also needs to be modified

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java (hunk 1 above)
Comment (IamMujuziMoses) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java]: SonarCloud is still producing errors

**Comment by IamMujuziMoses** [comment]
@zhangchunyang1024 @xingchun-chen how can I solve this sonarCloud error

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java [hunk 5]
```diff
@@ -90,12 +89,13 @@
...
-                               @RequestParam(value = "tenantName") String tenantName,
                                @RequestParam(value = "queueId") int queueId,
                                @RequestParam(value = "description", required = false) String description) throws Exception {
-        logger.info("login user {}, create tenant, tenantCode: {}, tenantName: {}, queueId: {}, desc: {}",
-                loginUser.getUserName(), tenantCode, tenantName, queueId, description);
-        Map<String, Object> result = tenantService.createTenant(loginUser, tenantCode, tenantName, queueId, description);
+        logger.info("login user {}, create tenant, tenantCode: {}, queueId: {}, desc: {}",
+                StringUtils.replaceNRTtoUnderline(loginUser.getUserName()),
+                StringUtils.replaceNRTtoUnderline(tenantCode), queueId,
+                StringUtils.replaceNRTtoUnderline(description));
```
Comment (zhuangchong) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java:97]: String userReplace = StringUtils.replaceNRTtoUnderline(loginUser.getUserName());
        String tenantCodeReplace = StringUtils.replaceNRTtoUnderline(tenantCode);
        String descReplace = StringUtils.replaceNRTtoUnderline(description);
        logger.info("login user {}, create tenant, tenantCode: {}, queueId: {}, desc: {}", userReplace, tenantCodeReplace, queueId, descReplace);

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java [hunk 6]
```diff
@@ -177,12 +175,13 @@ public Result queryTenantlist(@ApiIgnore @RequestAttribute(value = Constants.SES
...
-                               @RequestParam(value = "tenantName") String tenantName,
                                @RequestParam(value = "queueId") int queueId,
                                @RequestParam(value = "description", required = false) String description) throws Exception {
-        logger.info("login user {}, updateProcessInstance tenant, tenantCode: {}, tenantName: {}, queueId: {}, description: {}",
-                loginUser.getUserName(), tenantCode, tenantName, queueId, description);
-        Map<String, Object> result = tenantService.updateTenant(loginUser, id, tenantCode, tenantName, queueId, description);
+        logger.info("login user {}, updateProcessInstance tenant, tenantCode: {}, queueId: {}, description: {}",
+                StringUtils.replaceNRTtoUnderline(loginUser.getUserName()),
+                StringUtils.replaceNRTtoUnderline(tenantCode), queueId,
+                StringUtils.replaceNRTtoUnderline(description));
```
Comment (zhuangchong) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/TenantController.java:183]: String userReplace = StringUtils.replaceNRTtoUnderline(loginUser.getUserName());
        String tenantCodeReplace = StringUtils.replaceNRTtoUnderline(tenantCode);
        String descReplace = StringUtils.replaceNRTtoUnderline(description);
        logger.info("login user {}, create tenant, tenantCode: {}, queueId: {}, desc: {}", userReplace, tenantCodeReplace, queueId, descReplace);

**Review by zhuangchong — COMMENTED** [review]
Solve this sonarCloud 2 code smells

**Comment by zhuangchong** [comment]
Solve this sonarCloud 16.7% Coverage

Added in the includes tag of the pom file in the project root directory
<include>**/api/controller/TenantControllerTest.java</include>

**Comment by IamMujuziMoses** [comment]
the TenantControllerTest produces two `tenant not exists` errors

Screenshot (152)

Should I remove `<include>**/api/controller/TenantControllerTest.java</include>` from the pom.xml?!

**Comment by zhuangchong** [comment]
image

image


please change  Status.SUCCESS to Status.TENANT_NOT_EXIST

**Comment by dailidong** [comment]
I have updated related code

File: dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/TenantControllerTest.java [hunk 7]
```diff
@@ -165,7 +163,7 @@ public void testDeleteTenantById() throws Exception {
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON_UTF8))
                 .andReturn();
         Result result = JSONUtils.parseObject(mvcResult.getResponse().getContentAsString(), Result.class);
-        Assert.assertEquals(Status.SUCCESS.getCode(),result.getCode().intValue());
+        Assert.assertEquals(Status.TENANT_NOT_EXIST.getCode(),result.getCode().intValue());
```
Comment (dailidong) [inline dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/TenantControllerTest.java:166]: updated this error

Comment (IamMujuziMoses) [reply dolphinscheduler-api/src/test/java/org/apache/dolphinscheduler/api/controller/TenantControllerTest.java:166]: @dailidong yo needed to update the `testUpdateTenant` also

**Comment by zhuangchong** [comment]
LGTM.

**Comment by break60** [comment]
+1

**Review by dailidong — APPROVED** [review]
+1
good job,looking forward for your deep participation, thx

**Pull request merged**